test/acceptance/tests/controller/controller_namespaces_test.go

```diff
@@ -0,0 +1,166 @@
+
+	cases := []struct {
+		name                 string
+		destinationNamespace string
+		mirrorK8S            bool
+		secure               bool
+	}{
+		{
+			"single destination namespace (non-default)",
+			ConsulDestNS,
+			false,
+			false,
+		},
+		{
+			"single destination namespace (non-default); secure",
+			ConsulDestNS,
+			false,
+			true,
+		},
+		{
+			"mirror k8s namespaces",
+			KubeNS,
+			true,
+			false,
+		},
+		{
+			"mirror k8s namespaces; secure",
+			KubeNS,
+			true,
+			true,
+		},
+	}
+
+	for _, c := range cases {
+		t.Run(c.name, func(t *testing.T) {
+			ctx := suite.Environment().DefaultContext(t)
+
+			helmValues := map[string]string{
+				"global.enableConsulNamespaces": "true",
+				// todo: remove
+				"global.imageK8S": "lkysow/consul-k8s-dev:sep09-crd-ent6",
+				// todo: end
+				"controller.enabled":    "true",
+				"connectInject.enabled": "true",
+				// When mirroringK8S is set, this setting is ignored.
+				"connectInject.consulNamespaces.consulDestinationNamespace": c.destinationNamespace,
+				"connectInject.consulNamespaces.mirroringK8S":               strconv.FormatBool(c.mirrorK8S),
+
+				"global.acls.manageSystemACLs": strconv.FormatBool(c.secure),
+				"global.tls.enabled":           strconv.FormatBool(c.secure),
+			}
+
+			releaseName := helpers.RandomName()
+			consulCluster := framework.NewHelmCluster(t, helmValues, ctx, cfg, releaseName)
+
+			consulCluster.Create(t)
+
+			t.Logf("creating namespace %q", KubeNS)
+			out, err := helpers.RunKubectlAndGetOutputE(t, ctx.KubectlOptions(), "create", "ns", KubeNS)
+			if err != nil && !strings.Contains(out, "(AlreadyExists)") {
+				require.NoError(t, err)
+			}
+			helpers.Cleanup(t, cfg.NoCleanupOnFailure, func() {
+				helpers.RunKubectl(t, ctx.KubectlOptions(), "delete", "ns", KubeNS)
+			})
+
+			// Make sure that config entries are created in the correct namespace.
+			// If mirroring is enabled, we expect config entries to be created in the
+			// Consul namespace with the same name as their source
+			// Kubernetes namespace.
+			// If a single destination namespace is set, we expect all config entries
+			// to be created in that destination Consul namespace.
+			queryOpts := &api.QueryOptions{Namespace: KubeNS}
+			if !c.mirrorK8S {
+				queryOpts = &api.QueryOptions{Namespace: c.destinationNamespace}
+			}
+			consulClient := consulCluster.SetupConsulClient(t, c.secure)
+
+			// Test creation.
+			{
+				t.Log("creating service-defaults CRD")
+				helpers.RunKubectl(t, ctx.KubectlOptions(), "apply", "-n", KubeNS, "-f", "../fixtures/crds")
+				helpers.Cleanup(t, cfg.NoCleanupOnFailure, func() {
+					// NOTE: We're swallowing the delete error here because if
+					// the test ran successfully then this will already have
+					// been deleted.
+					helpers.RunKubectlAndGetOutputE(t, ctx.KubectlOptions(), "delete", "-n", KubeNS, "-f", "../fixtures/crds")
```

Comment:
Will this clean-up return an error if the delete section of this test executes successfully, given this resource has been deleted?